tests/test_api.rs

```diff
@@ -3386,6 +3386,63 @@ fn context_promise_hooks_partial() {
   }
 }
 
+#[test]
+fn context_nested_context() {
+  let _setup_guard = setup::parallel_test();
+  let isolate = &mut v8::Isolate::new(Default::default());
+  {
+    let scope = &mut v8::HandleScope::new(isolate);
+    let context = v8::Context::new(scope);
+    let scope = &mut v8::ContextScope::new(scope, context);
+
+    // Define a variable in the parent context
+    let global = {
+      let global = context.global(scope);
+      let variable_key = v8::String::new(scope, "variable").unwrap();
+      let variable_value = v8::String::new(scope, "value").unwrap();
+      global.set(scope, variable_key.into(), variable_value.into());
+      v8::Global::new(scope, global)
+    };
+
+    // Creates a child context
+    {
+      let security_token = context.get_security_token(scope);
+      let templ = v8::ObjectTemplate::new(scope);
+      let global = v8::Local::new(scope, global);
+      templ.set_named_property_handler(
+        v8::NamedPropertyHandlerConfiguration::new()
+          .getter(
+            |scope: &mut v8::HandleScope,
+             key: v8::Local<v8::Name>,
+             args: v8::PropertyCallbackArguments,
+             mut rv: v8::ReturnValue| {
+              let obj = v8::Local::<v8::Object>::try_from(args.data()).unwrap();
+              if let Some(val) = obj.get(scope, key.into()) {
+                rv.set(val);
+              }
+            },
+          )
+          .data(global.into()),
+      );
+      let child_context = v8::Context::new_from_template(scope, templ);
+
+      // Without the security context, the variable can not be shared
```

Comment:
This statement isn't checked in the test - I think it would be good to run the same source code below and see that it does throw.